core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -84,21 +86,33 @@ public static AwsCredentialsProviderChain of(AwsCredentialsProvider... awsCreden
         return builder().credentialsProviders(awsCredentialsProviders).build();
     }
 
+    /**
+     * Create an AWS credentials provider chain with default configuration that checks the given credential providers.
+     * @param awsCredentialsProviders The credentials providers that should be checked for credentials, in the order they should
+     *                                be checked.
+     * @return A credential provider chain that checks the provided credential providers in order.
+     */
+    public static AwsCredentialsProviderChain of(IdentityProvider<? extends AwsCredentialsIdentity>... awsCredentialsProviders) {
+        return builder().credentialsProviders(awsCredentialsProviders).build();
+    }
+
     @Override
     public AwsCredentials resolveCredentials() {
         if (reuseLastProviderEnabled && lastUsedProvider != null) {
-            return lastUsedProvider.resolveCredentials();
+            // TODO: Exception handling for join?
+            return CredentialUtils.toCredentials(lastUsedProvider.resolveIdentity().join());
         }
 
         List<String> exceptionMessages = null;
-        for (AwsCredentialsProvider provider : credentialsProviders) {
+        for (IdentityProvider<? extends AwsCredentialsIdentity> provider : credentialsProviders) {
             try {
-                AwsCredentials credentials = provider.resolveCredentials();
+                // TODO: Exception handling for join?
+                AwsCredentialsIdentity credentials = provider.resolveIdentity().join();
```

Comment:
Should we change the variable name `credentials` to reflect something related identity like identityCredentials ?

Reply:
See other similar comment.